app/src/main/java/org/eyeseetea/malariacare/data/sync/importer/PullController.java

```diff
@@ -17,62 +17,91 @@
-import org.eyeseetea.malariacare.database.model.OrgUnit;
-import org.eyeseetea.malariacare.database.model.Program;
-import org.eyeseetea.malariacare.database.model.Question;
-import org.eyeseetea.malariacare.database.model.QuestionOption;
-import org.eyeseetea.malariacare.database.utils.PopulateDB;
-import org.eyeseetea.malariacare.database.utils.PreferencesState;
-import org.eyeseetea.malariacare.sdk.SdkController;
-import org.eyeseetea.malariacare.sdk.SdkPullController;
-import org.eyeseetea.malariacare.sdk.SdkQueries;
+import org.eyeseetea.malariacare.data.database.model.Option;
+import org.eyeseetea.malariacare.data.database.model.OrgUnit;
+import org.eyeseetea.malariacare.data.database.model.Program;
+import org.eyeseetea.malariacare.data.database.model.Question;
+import org.eyeseetea.malariacare.data.database.model.QuestionOption;
+import org.eyeseetea.malariacare.data.database.utils.PopulateDB;
+import org.eyeseetea.malariacare.data.database.utils.PreferencesState;
+import org.eyeseetea.malariacare.data.remote.SdkController;
+import org.eyeseetea.malariacare.data.remote.SdkPullController;
+import org.eyeseetea.malariacare.data.remote.SdkQueries;
+import org.eyeseetea.malariacare.data.sync.importer.models.DataValueExtended;
+import org.eyeseetea.malariacare.data.sync.importer.models.EventExtended;
+import org.eyeseetea.malariacare.data.sync.importer.models.OrganisationUnitExtended;
+import org.eyeseetea.malariacare.data.sync.importer.models.ProgramExtended;
+import org.eyeseetea.malariacare.domain.boundary.IPullController;
 import org.hisp.dhis.client.sdk.models.program.ProgramType;
 
+import java.io.IOException;
 import java.util.Calendar;
 import java.util.Date;
 import java.util.List;
 
-/**
- * A static controller that orchestrate the pull process
- * Created by arrizabalaga on 4/11/15.
- */
-public class PullController {
-    public static final int MAX_EVENTS_X_ORGUNIT_PROGRAM = 4800;
-    public static final int NUMBER_OF_MONTHS = 0;
-    private static PullController instance;
-    private final String TAG = ".PullController";
-    /**
-     * Context required to i18n error messages while pulling
-     */
-    private Context context;
+public class PullController implements IPullController {
+    private static String TAG = "PullController";
+    //public static final int MAX_EVENTS_X_ORGUNIT_PROGRAM = 4800;
 
-    /**
-     * Constructs and register this pull controller to the event bus
-     */
-    PullController() {
+    private Context mContext;
+
+    public PullController(Context context) {
+        mContext = context;
     }
 
-    /**
-     * Singleton constructor
-     */
-    public static PullController getInstance() {
-        if (instance == null) {
-            instance = new PullController();
+    public void pull(Callback callback) {
+        Log.d(TAG, "Starting PULL process...");
+        try {
+
+            populateMetadataFromCsvs();
+
+            callback.onComplete();
+/*            //clear flags
```

Comment:
@xurxodev i don't understand, why it is commented?. Maybe should have a todo instead of comment it? What do you think?